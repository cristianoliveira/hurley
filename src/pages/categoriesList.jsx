import React, { Component } from 'react';
import { Link } from 'react-router';

class CategoriesList extends Component {
  render() {
    return (
      <div>
        <h1> Lista de Categorias </h1>
      </div>
    );
  }
}

export default CategoriesList;
