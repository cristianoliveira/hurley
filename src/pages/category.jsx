import React, { Component } from 'react';
import { Link } from 'react-router';
import Helmet from 'react-helmet';
import ChannelCard from './../components/ChannelCard'
import FeedcastLoader from './../components/FeedcastLoader'
import feedcastApi from 'feedcast-client';
import helpers from './../scripts/helpers'

class Category extends Component {

  constructor(props) {
    super(props);

    let { lc } = helpers.localize(this)

    this.state = {
      channels : [],
      icon: null,
      slug: null,
      title: null,
      populated: false,
      lc
    }
  }


  componentWillUnmount(){
    this._isMounted = false
  }

  componentDidMount() {
    this._isMounted = true
    this.updatePage(this.props)
  }


  updatePage(props){
    const { page } = props.params
    const slug = page && page.length > 0 ? page : ''
    this.setState({ populated: false }, ()=>{
      feedcastApi
        .getEpisodesByCategory({slug})
        .then( data => {
          this.setState({
            populated: true,
            ...data
          });
        })
    });
  }



  listChannels(){
    let { channels } = this.state;
    return channels
            .filter(c => c.listed)
            .map( (c, n) => (<ChannelCard key={n} data={c}/>));
  }


  componentWillUpdate(nextProps) {
    if( this.props.params.page != nextProps.params.page)
      this.updatePage(nextProps);
  }



  render() {
    const channelList = this.listChannels()

    return this.state.populated ? (
      <div>
        <Helmet
          title={`Feedcast`}
          meta={[
            {property: 'og:title',
            content: `Feedcast ${ this.state.populated ? '| ' + this.state.title : '' }`},
          ]} />
        <h1>{helpers.translate(this.state.title)} </h1>
        <div className="feedcast__channel-list feedcast__channel-list--byCategory">
          {channelList}
        </div>
      </div>
    ) : (<FeedcastLoader/>);
  }
}

export default Category;
