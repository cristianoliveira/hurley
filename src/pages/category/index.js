import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestCategoryInfo } from 'app/actions/categories';
import FeedcastLoader from 'app/components/FeedcastLoader';
import Category from './Category';

class CategoryContainer extends PureComponent {
  static defaultProps = {
    categories: [],
    isFetching: true,
    params: {},
  }

  fetchData(slug) {
    this.props.dispatch(requestCategoryInfo(slug));
  }

  render() {
    if (this.props.isFetching) {
      return <FeedcastLoader />;
    }

    const category = this.props.categories[0] || {};
    if (category.slug !== this.props.match.params.slug) {
      this.fetchData(this.props.match.params.slug);
    }

    return <Category {...this.props} category={this.props.categories[0]} />;
  }
}

const mapStateToProps = (state) => state.categories;

export { Category };
export default connect(mapStateToProps)(CategoryContainer);
