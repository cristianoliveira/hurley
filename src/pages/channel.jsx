import React, { Component } from 'react';
import { Link } from 'react-router';
import Helmet from 'react-helmet';

import feedcastApi from './../scripts/feedcastApi'
import helpers from './../scripts/helpers'

import './../styles/channel.sass'
import ChannelEpisodes from './../components/ChannelEpisodes.jsx'
import FeedcastLoader from './../components/FeedcastLoader.jsx'


class Channel extends Component {
  constructor(){
    super();

    this.state = {
      categories : [],
      description: '',
      feed_url: '',
      image_url: '',
      listed: false,
      slug: '',
      synchronization_status: '',
      synchronization_status_message: '',
      title: '',
      uuid: '',
      populated: false
    }
  }

  componentDidMount() {
    this._isMounted = true
  }

  componentWillMount() {
    this._isMounted = false
    const { uuid } = this.props.params
    feedcastApi
      .getChannelInfo({ uuid })
      .then(data => {
        data.populated = true
        this.setState(data)
      })
  }



  render() {
    let { categories, description,
      image_url, title, uuid } = this.state

    let { page } = this.props.params

    categories = categories.map( (i,n) => (
      <Link key={n} to={`/category/${i.slug}`}>
        <i className={`fa fa-${i.icon}`}></i> {helpers.translate(i.title)}
      </Link>
    ));

    let metaTitle = this.state.populated ? `| ${this.state.title}` : ``

    return this.state.populated ? (
      <div className="feedcast__channelPage">
        <Helmet
          title={`Feedcast ${metaTitle}`}
          meta={[
            {property: 'og:title',
            content: `Feedcast ${metaTitle}`},
          ]} />
        <div className="feedcast__channelInfo">
          <div className="feedcast__channelInfo-header">
            <img className="feedcast__channelInfo-img" src={image_url}/>
            <h1 className="feedcast__channelInfo-title">{title}</h1>
          </div>
          <div className="feedcast__channelInfo-body">
            <p className="feedcast__channelInfo-description">{description}</p>
            <div className="feedcast__channelInfo-categories">{categories}</div>
          </div>
        </div>
        <ChannelEpisodes data={{uuid, page}}/>
      </div>
    ) : (
      <FeedcastLoader />
    );
  }
}

export default Channel;
