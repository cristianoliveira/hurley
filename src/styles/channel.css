.feedcast__channelInfo,
.feedcast__channelEpisodes {
  background-color: white;
}
.feedcast__channelInfo-header{
  padding: 10px;
  box-sizing: border-box;
  border-bottom: 1px solid var(--lighter-gray2);
  background-color: var(--darker-blue);
  height: 100px;
  position: relative;
}
.feedcast__channelInfo-body{
  padding: 10px;
  padding-top: 40px;
  box-sizing: border-box;
}
.feedcast__channelInfo-img,
.feedcast__channelInfo-title{
  display: inline-block;
}
.feedcast__channelInfo-img{
  width: 100px;
  height: 100px;
  margin-right: 10px;
  float: left;
  border: 1px solid white;
  position: relative;
  transform: translateY(20px);
}
.feedcast__channelInfo-img:before {
  content:"\f2ce";
  position: absolute;
  width: 100px;
  height: 100px;
  display: block;
  background-color: var(--lighter-blue);
  color: white;
  font-family: FontAwesome;
  font-size: 60px;
  text-align: center;
  line-height: 100px;
  border-bottom-right-radius: 5px;
  border-bottom-left-radius: 5px;
}
.feedcast__channelInfo-title{
  color: white;
  margin:0;
  position: absolute;
  bottom:5px;
  left:120px;
  font-size: 1.3em;
}
.feedcast__channelInfo-categories a {
  text-decoration: none;
  display: inline-block;
  margin-right: 10px;
  margin-bottom: 7px;
  background-color: var(--orange);
  color: white;
  padding: 5px 10px;
  border-radius: 5px;
  font-size: 12px;
  font-weight: bolder;
}
.feedcast__channelInfo-description{
  text-align: justify;
  text-justify: inter-word;
}

.feedcast__channelEpisodes {
  padding: 10px;
  margin-top: 20px;
  box-sizing: border-box;
}
.feedcast__channelEpisode {
  /*background-color: var(--lighter-gray);*/
  margin-bottom: 10px;
  padding-bottom: 10px;
  border-bottom: 1px solid var(--lighter-gray2)
}

.feedcast__channelEpisode-info-wrapper,
.feedcast__channelEpisode-button-wrapper {
  display: inline-block;
}

.feedcast__channelEpisode-button-wrapper {
  width: 100px;
  min-height: 80px;
  float: left;
  box-sizing: border-box;
  padding: 10px;
}

.feedcast__channelEpisode-button-wrapper button{
  width:60px;
  height:60px;
  margin-left: 50%;
  transform: translateX(-50%);
  border-radius: 50%;
  border: 0px;
  background-color: white;
  border: 1px solid var(--darker-blue);
  color: var(--darker-blue);
  font-size: 20px;
  cursor: pointer;
  transition: .4s ease all;
}
.feedcast__channelEpisode-button-wrapper button:hover{
  background-color: var(--darker-blue);
  color: white;
}

.feedcast__channelEpisode-info-wrapper {
  width: calc(100% - 100px);
  box-sizing: border-box;
  padding: 10px;
}
.feedcast__channelEpisode-summary {
  margin-bottom: 0px;
}
.feedcast__channelEpisode-summary--see-less {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}


.feedcast__channelEpisode-info-wrapper h4 {
  margin-top: 0px;
}
.feedcast__channelEpisode-info-wrapper span {
  color: var(--orange);
  border: 1px solid var(--orange);
  background-color: white;
  padding: 0px 5px;
  font-weight: lighter;
  border-radius: 3px;
  float: right;
  margin-top: 20px;
}
.feedcast__channelEpisode-btn-see-more{
  background-color: transparent;
  border: none;
  text-decoration:underline;
  color: var(--lighter-blue);
  cursor: pointer;
}

@media ( min-width: 960px){
  .feedcast__channelInfo,
  .feedcast__channelEpisodes {
    display: inline-block;
    float: left;
    margin:0px;
  }
  .feedcast__channelInfo{
    width: 400px;
    position: fixed;
  }
  .feedcast__channelEpisodes {
    width: calc(100% - 410px);
    margin-left: 410px;
  }
}
