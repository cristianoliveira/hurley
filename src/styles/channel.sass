@import 'constants'

.feedcast__channelInfo,
.feedcast__channelEpisodes
  background-color: white;

.feedcast__channelInfo-header
  padding: 10px;
  box-sizing: border-box;
  border-bottom: 1px solid $lighter-gray2;
  background-color: $darker-blue;
  height: 100px;
  position: relative;

.feedcast__channelInfo-body
  padding: 10px;
  padding-top: 40px;
  box-sizing: border-box;

.feedcast__channelInfo-img,
.feedcast__channelInfo-title
  display: inline-block;

.feedcast__channelInfo-img
  width: 100px;
  height: 100px;
  margin-right: 10px;
  float: left;
  border: 1px solid white;
  position: relative;
  transform: translateY(20px);

.feedcast__channelInfo-img:before
  content: "\f2ce";
  position: absolute;
  width: 100px;
  height: 100px;
  display: block;
  background-color: $lighter-blue;
  color: white;
  font-family: FontAwesome;
  font-size: 60px;
  text-align: center;
  line-height: 100px;
  border-bottom-right-radius: 5px;
  border-bottom-left-radius: 5px;

.feedcast__channelInfo-title
  color: white;
  margin: 0;
  position: absolute;
  bottom: 5px;
  left: 120px;
  font-size: 1.3em;

.feedcast__channelInfo-categories a
  text-decoration: none;
  display: inline-block;
  margin-right: 10px;
  margin-bottom: 7px;
  background-color: $orange;
  color: white;
  padding: 5px 10px;
  border-radius: 5px;
  font-size: 12px;
  font-weight: bolder;

.feedcast__channelInfo-description
  text-align: justify;
  text-justify: inter-word;


.feedcast__channelEpisodes
  padding: 10px;
  margin-top: 20px;
  box-sizing: border-box;

.feedcast__channelEpisodes .feedcast__pagination
  margin-top: 10px

.feedcast__channelEpisode
  padding-bottom: 10px;
  padding-top: 10px;
  border-bottom: 1px solid $lighter-gray;
  transition: ease all 1s

  &:hover
    background-color: #f0f0f0

    .feedcast__channelEpisode-info-wrapper button
      visibility: visible

.feedcast__channelEpisode-button-wrapper
  display: inline-block;


.feedcast__channelEpisode-button-wrapper
  width: 50px;
  padding-left: 10px;
  float: left;
  box-sizing: border-box;


.feedcast__channelEpisode-button-wrapper button
  width: 25px;
  height: 25px;
  border-radius: 50%;
  border: 0px;
  background-color: white;
  border: 1px solid $darker-blue;
  color: $darker-blue;
  font-size: 10px;
  cursor: pointer;
  transition: .4s ease all;

.feedcast__channelEpisode-button-wrapper button:hover
  background-color: $darker-blue;
  color: white;


.feedcast__channelEpisode-info-wrapper
  display: inline-block;
  width: calc(100% - 50px)
  box-sizing: border-box;

.feedcast__channelEpisode-summary
  margin-bottom: 0px;

.feedcast__channelEpisode-summary--see-less
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;



.feedcast__channelEpisode-info-wrapper h4
  margin: 0px;
  width: calc(100% - 150px);
  display: inline-block
  font-weight: lighter
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  line-height: 1em;
  transform: translateY(3px);


.feedcast__channelEpisode-info-wrapper button
  visibility: hidden
  border: 0px
  background-color: transparent
  display: inline-block
  width: 30px
  padding: 0px
  margin: 0px
  margin-right: 20px
  height: 24px
  line-height: 24px
  cursor: pointer
  i
    text-align: center
    display: block

.feedcast__channelEpisode-info-wrapper span
  color: $orange;
  border: 1px solid $orange;
  background-color: white;
  padding: 0px 5px;
  font-weight: lighter;
  border-radius: 3px;
  width: 80px;
  text-align: center
  display: inline-block

.feedcast__channelEpisode-btn-see-more
  background-color: transparent;
  border: none;
  text-decoration: underline;
  color: $lighter-blue;
  cursor: pointer;

.feedcast__channelPage
  position: relative
  max-width: 960px
  left: 50%
  transform: translateX(-50%)

@media ( min-width: 960px)
  .feedcast__channelPage
    display: block
  .feedcast__channelInfo,
  .feedcast__channelEpisodes
    display: block;
    margin: 0px;

  .feedcast__channelInfo
    width: 100%;

  .feedcast__channelEpisodes
    width: 100%;


