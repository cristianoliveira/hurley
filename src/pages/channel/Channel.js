import React, { Component } from 'react';
import { Link } from 'react-router';
import Helmet from 'react-helmet';

import helpers from 'app/scripts/helpers'

import 'app/styles/channel.sass'
import ChannelEpisodes from 'app/components/ChannelEpisodes'


class Channel extends Component {
  renderCategories() {
    return this.props.channel.categories.map( (i,n) => (
      <Link key={n} to={`/category/${i.slug}`}>
        <i className={`fa fa-${i.icon}`}></i> {helpers.translate(i.title)}
      </Link>
    ));
  }

  render() {
    if (!this.props.channel) {
      return null;
    }

    const {
      description,
      image_url,
      title,
    } = this.props.channel;

    const metaTitle = this.props.channel ? `| ${title}` : ``

    return (
      <div className="feedcast__channelPage">
        <Helmet
          title={`Feedcast ${metaTitle}`}
          meta={[
            {property: 'og:title',
            content: `Feedcast ${metaTitle}`},
          ]} />
        <div className="feedcast__channelInfo">
          <div className="feedcast__channelInfo-header">
            <img className="feedcast__channelInfo-img" alt={title} src={image_url}/>
            <h1 className="feedcast__channelInfo-title">{title}</h1>
          </div>
          <div className="feedcast__channelInfo-body">
            <p className="feedcast__channelInfo-description">{description}</p>
            <div className="feedcast__channelInfo-categories">
              { this.renderCategories.bind(this) }
            </div>
          </div>
        </div>
        <ChannelEpisodes {...this.props} />
      </div>
    );
  }
}

export default Channel;
