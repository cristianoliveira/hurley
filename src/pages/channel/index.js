import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { requestChannelInfo } from 'app/actions/channels';
import FeedcastLoader from 'app/components/FeedcastLoader';
import Channel from './Channel';

class ChannelContainer extends PureComponent {

  static defaultProps = {
    episodes: [],
    isFetching: true,
    params: {},
  }

  componentDidMount() {
    const uuid = this.props.params.uuid;
    if (uuid !== this.props.uuid) {
      this.fetchData(uuid);
    }
  }

  fetchData(channelID) {
    this.props.dispatch(requestChannelInfo(channelID));
  }

  render() {
    if (this.props.isFetching) {
      return <FeedcastLoader />;
    }

    return <Channel {...this.props} channel={this.props.channels[0]} />;
  }
}

const mapStateToProps = (state) => state.channels;

export { Channel };
export default connect(mapStateToProps)(ChannelContainer);
