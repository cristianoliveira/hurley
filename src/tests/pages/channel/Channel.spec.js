import React from 'react';
import renderer from 'react-test-renderer';
import { withStore } from 'app/tests/helpers';

import { Channel } from 'app/pages/channel';

describe('Channel', () => {
  describe('when has no data', () => {
    it("renders the no episode", () => {
      const component = renderer.create(withStore(<Channel />));
      expect(component).toMatchSnapshot();
    })
  });

  describe('when has a list of channels', () => {
    const state = {
      channels: [
        {
          uuid: "asdasdadasda",
          slug: "pensando-rpg",
          title: "Pensando RPG",
          description: "",
          feed_url: "http://feeds.soundcloud.com/users/soundcloud:users:257525576/sounds.rss",
          image_url: "https://img.feedcast.io/http%3A%2F%2Fi.imgur.com%2FmYRfJk7.png/300/300",
          listed: true,
          synchronization_status: "success",
          synchronization_status_message: "",
          categories: [],
        }
      ],
      page: 1,
      perPage: 130,
      total: 1000,
    }

    it('render the channels', () => {
      const component = renderer.create(withStore(<Channel {...state} />));
      expect(component).toMatchSnapshot();
    });
  });
})
