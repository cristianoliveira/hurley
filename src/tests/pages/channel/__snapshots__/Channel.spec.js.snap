// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel when has a list of channels render the channels 1`] = `null`;

exports[`Channel when has no data renders the no episode 1`] = `null`;
